sdk/python/feast/job_service.py

```diff
@@ -117,6 +126,22 @@ def StartStreamToOnlineIngestionJob(
         feature_table = self.client.get_feature_table(
             request.table_name, request.project
         )
+
+        if self.client._config.getboolean(CONFIG_JOB_SERVICE_ENABLE_CONTROL_LOOP):
+            # If the control loop is enabled, return existing stream ingestion job id instead of starting a new one
+            params = get_stream_to_online_ingestion_params(
+                self.client, request.project, feature_table, []
+            )
+            job_hash = params.get_job_hash()
+            jobs_by_hash = list_jobs_by_hash(self.client, include_terminated=False)
+            if job_hash not in jobs_by_hash:
+                raise RuntimeError(
+                    "Feast Job Service has control loop enabled, but couldn't find the existing stream ingestion job for the given FeatureTable"
```

Comment:
Can we use lower case for concepts as a rule of thumb? `feature table`.

Reply:
Also, are we really going to raise an exception if the job doesn't exist when they run a `start` method? Shouldn't they be able to idempotently run this method?

Reply:
Can you describe the race condition? The idea is that if the job doesn't already exist and the control loop is working in the background, it'll probably start this job soon. So here we don't start the spark job, hence no race condition. We would have the race condition if we tried to start the stream ingestion job here, since the control loop could also be trying to do the same at the same time.

Decided to throw an exception instead of returning response with empty job_id, because even if we return the response the `job.get_status()` and other methods will fail anyways, since job_id is empty. And that error will be more difficult to debug to users than the explicit exception.